### PR TITLE
HAAR-438 Correct use account status and allow filtering

### PR DESCRIPTION
**Pull request by Mjwillis**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisuserrolesapi/service/UserService.kt [hunk 1]
```diff
@@ -625,7 +585,8 @@ class UserService(
     val userPersonDetail = UserPersonDetail(
       username = username.uppercase(),
       staff = staffAccount,
-      type = getUsageType(admin)
+      type = getUsageType(admin),
+      accountDetail = null
```
Comment (Mjwillis) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/nomisuserrolesapi/service/UserService.kt:589]: Don't want in the real DB to insert data into DBA_USERS as its a view

File: src/test/kotlin/uk/gov/justice/digital/hmpps/nomisuserrolesapi/helper/EntityDataLoader.kt [hunk 2]
```diff
@@ -181,7 +184,8 @@ fun defaultPerson(): UserPersonDetail {
   return UserPersonDetail(
     username = "tony",
     staff = Staff(firstName = "John", lastName = "Smith", status = "ACTIVE"),
-    type = UsageType.GENERAL
+    type = UsageType.GENERAL,
+    accountDetail = AccountDetail(accountStatus = AccountStatus.OPEN.desc, profile = AccountProfile.TAG_GENERAL.name)
```
Comment (Mjwillis) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/nomisuserrolesapi/helper/EntityDataLoader.kt:188]: Done in tests only

File: src/test/kotlin/uk/gov/justice/digital/hmpps/nomisuserrolesapi/resource/UserAccountResourceIntTest.kt [hunk 3]
```diff
@@ -119,11 +117,11 @@ class UserAccountResourceIntTest : IntegrationTestBase() {
         .expectBody()
         .jsonPath("generalAccount.username").isEqualTo("GENERALUSER1")
         .jsonPath("generalAccount.activeCaseload.id").isEqualTo("WWI")
-        .jsonPath("generalAccount.active").isEqualTo("true")
+        .jsonPath("generalAccount.active").isEqualTo("false")
```
Comment (Mjwillis) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/nomisuserrolesapi/resource/UserAccountResourceIntTest.kt:120]: status would now be inactive for new accounts

File: src/test/kotlin/uk/gov/justice/digital/hmpps/nomisuserrolesapi/service/UserServiceTest.kt [hunk 4]
```diff
@@ -84,87 +81,93 @@ internal class UserServiceTest {
...
         AccountProfile.TAG_GENERAL.name,
         LocalDateTime.now().plusMinutes(1)
       )
-      assertThat(userService.isCredentialsNonExpired(accountDetailPasswordValid)).isTrue
+      assertThat(accountDetailPasswordValid.isCredentialsNonExpired()).isTrue
     }
 
     @Test
     fun `isEnabled`() {
-      // Given
```
Comment (Mjwillis) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/nomisuserrolesapi/service/UserServiceTest.kt:130]: Not needed

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisuserrolesapi/jpa/specification/UserSpecification.kt [hunk 5]
```diff
@@ -93,7 +96,9 @@ class UserSpecification(private val filter: UserFilter) : Specification<UserPers
       }
 
     fun statusMatch(status: UserStatus): Predicate? =
-      status.databaseStatus()?.let { equal(get(UserPersonDetail::staff).get(Staff::status), it) }
+      status.databaseStatus()?.let {
+        inList(get(UserPersonDetail::accountDetail).get(AccountDetail::accountStatus), it)
```
Comment (Mjwillis) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/nomisuserrolesapi/jpa/specification/UserSpecification.kt:100]: contains one of the "ACTIVE" or "INACTIVE" statuses

**Review by andymarke — APPROVED** [review]
LGTM

**Pull request merged**